lib/clouds/libcloud_common.py

```diff
@@ -46,162 +46,108 @@ class LibcloudCmds(CommonCloudFunctions) :
                 1. Cloud is named "FOO"
                 2. two SSH keys, identified by "a" and "b"
                 3. The configuration file would look like this:
-
                  [USER-DEFINED : CLOUDOPTION_FOO ]
                  FOO_KEY_NAME = a,b # identifiers used by cloud-init to install the public key
                  FOO_SSH_KEY_NAME = path/to/private/key/for/cbtool
-
             Using libcloud, we will then go and lookup those keys and use them at instance creation time.
-
             NOTE: FOO_SSH_KEY_NAME != FOO_SSH_KEY_NAME. The first one is the private key used by cbtool,
                   and the second one is the public key.
-
             NOTE: If your cloud doesn't support cloud-init, then you will need to make sure the VM image is prepared in advance
                   with the public key that corresponds to FOO_SSH_KEY_NAME
-
     @use_cloud_init: (Optional, Default False)
             NOTE: If your cloud doesn't support cloud-init, you will need to make sure your images have baked in the SSH public key on your own.
-
-    @use_volumes: (Optional, Default False)
-
-            Leave this as false if you're not interested in block storage / volume support, or you're cloud doesn't support it.
-
     @use_sizes: (Optional, Default True)
             Use the VM image sizes as listed by your cloud.
-
     @use_locations: (Optional, Default True)
             Use the Regions as listed by your cloud.
-
     @verify_ssl: (Optional, Default True)
             Whether or not to have libcloud verify SSL certificates when communicating with the cloud
-
     @tldomain: (Optional, Default False)
             The FQDN of your cloud. None if false.
-
     @extra: (Optional, Default empty)
             Extra fixed parameters to be used by libcloud that we don't know about, which get passed at instance creation time.
-
     '''
 
     @trace
-    def __init__ (self, pid, osci, expid = None, description = "OverrideMe", num_credentials = 2, use_ssh_keys = False, use_cloud_init = False, use_volumes = False, use_locations = True, use_sizes = True, tldomain = False, verify_ssl = True, extra = {}) :
+    def __init__ (self, pid, osci, expid = None, provider = "OverrideMe", \
+                  num_credentials = 2, use_ssh_keys = False, \
+                  use_cloud_init = False,  use_locations = True, \
+                  use_sizes = True, tldomain = False, verify_ssl = True, extra = {}) :
+        '''
+        TBD
+        '''
         CommonCloudFunctions.__init__(self, pid, osci, expid)
         self.access_url = False
         self.ft_supported = False
         self.current_token = 0
         self.cache_mutex = threading.Lock()
         self.token_mutex = threading.Lock()
-        self.description = description
+        self.provider = provider
         self.num_credentials = num_credentials
         self.use_ssh_keys = use_ssh_keys
         self.use_cloud_init = use_cloud_init
-        self.use_volumes = use_volumes
```

Comment:
So, I removed the "use_volumes", due to the existence of the "fake" volume functions, but of course its entirely open for discussion...